ToolBox/BovespaDownloader/BovespaDataDownloader.cs

```diff
@@ -0,0 +1,449 @@
+                                    {
+                                        Log.Error(x + " is out of place!");
+                                        return false;
+                                    }
+                                });
+                        }
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                return new DirectoryInfo(_inputDirectory).EnumerateFiles("*.zip").Where(x => x.Length > 0 && !x.Name.Contains("COTAHIST_A") && !x.Name.Contains("OFER")).Select(x => x.Name);
+            }
+        }
+        
+        /// <summary>
+        /// Get ticks for the specified symbool from a given file.
+        /// If we do not have the file locally we download it.
+        /// </summary>
+        /// <param name="symbol">The requested symbol</param>
+        /// <param name="date">The requested date</param>
+        /// <returns>An enumerable of ticks</returns>
+        private IEnumerable<BaseData> GetTickData(Symbol symbol, string file)
+        {
+            var localfile = String.Format("{0}/{1}", _inputDirectory, file);
+
+            #region Download file from FTP site if it does not exist locally
+            if (!File.Exists(localfile) || new FileInfo(localfile).Length == 0)
+            {
+                try
+                {
+                    var request = (FtpWebRequest)WebRequest.Create(String.Format("{0}/{1}", _ftpsite, file));
+                    request.Credentials = new NetworkCredential("anonymous", "me@home.com");
+                    request.Method = WebRequestMethods.Ftp.DownloadFile;
+
+                    using (var response = (FtpWebResponse)request.GetResponse())
+                    {
+                        using (var responseStream = response.GetResponseStream())
+                        {
+                            var readCount = 0;
+                            var buffer = new byte[2048];
+
+                            // Create new local file
+                            using (var newFile = new FileStream(localfile, FileMode.Create))
+                            {
+                                while ((readCount = responseStream.Read(buffer, 0, buffer.Length)) > 0)
+                                    newFile.Write(buffer, 0, readCount);
+                            }
+                        }
+                    }
+                }
+                catch (Exception) { }
+            }
+            #endregion
+
+            if (!File.Exists(localfile))
+            {
+                Console.WriteLine("Do not have nor could download " + file);
+            }
+            else
+            {
+                ZipFile zip;
+                var isBid = localfile.Contains("OFER_CPA");
+                var isAsk = localfile.Contains("OFER_VDA");
+                var tickType = localfile.Contains("NEG") ? TickType.Trade : TickType.Quote;
+
+                using (var toplevelreader = Compression.Unzip(localfile, out zip))
```

Comment:
I think when we use this method we need to manually dispose of the `out zip` parameter.